dragonfly_energy/properties/building.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+"""Building Energy Properties."""
+from honeybee_energy.programtype import ProgramType
+from honeybee_energy.constructionset import ConstructionSet
+from honeybee_energy.idealair import IdealAirSystem
+
+from honeybee_energy.lib.constructionsets import generic_construction_set
+
+
+class BuildingEnergyProperties(object):
+    """Energy Properties for Dragonfly Building.
+
+    Properties:
+        * host
+        * construction_set
+    """
+
+    __slots__ = ('_host', '_construction_set')
+
+    def __init__(self, host, construction_set=None):
+        """Initialize Building energy properties.
+
+        Args:
+            host: A dragonfly_core Building object that hosts these properties.
+            construction_set: A honeybee ConstructionSet object to specify all
+                default constructions for the Faces of the Building. If None, the
+                Building will use the honeybee default construction set, which is not
+                representative of a particular building code or climate zone.
+                Default: None.
+        """
+        self._host = host
+        self.construction_set = construction_set
+
+    @property
+    def host(self):
+        """Get the Building object hosting these properties."""
+        return self._host
+
+    @property
+    def construction_set(self):
+        """Get or set the Building ConstructionSet object.
+
+        If not set, it will be the Honeybee default generic ConstructionSet.
+        """
+        if self._construction_set is not None:  # set by the user
+            return self._construction_set
+        else:
+            return generic_construction_set
+
+    @construction_set.setter
+    def construction_set(self, value):
+        if value is not None:
+            assert isinstance(value, ConstructionSet), \
+                'Expected ConstructionSet. Got {}'.format(type(value))
+            value.lock()   # lock in case construction set has multiple references
+        self._construction_set = value
+
+    def averaged_program_type(self, name=None, timestep_resolution=1):
```

Comment:
This is an interesting metric. Do we have a specific use case for this or you just thought it would be interesting to get an overall view the program types in one building?

Reply:
There's no pressing use case right now but I think it will be really useful down the line for informing the `dragonfly-uwg` extension. The UWG can only take one "ProgramType" definition per building type and so the ability to get an averaged ProgramType across a building will be used by any `to_uwg` method.